From 0e0b0eb3e5b3c046971ff69aae1ca69c1450f5b4 Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Mon, 27 Mar 2023 10:17:37 -0700
Subject: [PROPOSED] =?UTF-8?q?Revert=202023b=E2=80=99s=20data=20changes?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

* NEWS: Mention this.
* asia (Lebanon): Revert to 2023a data.  Add commentary.
* tz-link.html: Warn further about confusion.
---
 NEWS         | 13 +++++++++----
 asia         | 39 ++++++++++++++++++++++++++++++++++-----
 tz-link.html |  6 +++---
 3 files changed, 46 insertions(+), 12 deletions(-)

diff --git a/NEWS b/NEWS
index 9b235a29..fe0e809b 100644
--- a/NEWS
+++ b/NEWS
@@ -1,14 +1,19 @@
 News for the tz database
 
-Release 2023b - 2023-03-23 19:50:38 -0700
+Unreleased, experimental changes
 
-  Briefly:
-    Lebanon delays the start of DST this year.
+  Changes to past and future timestamps
+
+    Model Lebanon's DST chaos by reverting data to tzdb 2023a.
+    (Thanks to Jad Baz for the heads-up.)
+
+
+Release 2023b - 2023-03-23 19:50:38 -0700
 
   Changes to future timestamps
 
     This year Lebanon springs forward April 20/21 not March 25/26.
-    (Thanks to Saadallah Itani.)
+    (Thanks to Saadallah Itani.)  [This was reverted in 2023c.]
 
 
 Release 2023a - 2023-03-22 12:39:33 -0700
diff --git a/asia b/asia
index dd06a5fd..180b3992 100644
--- a/asia
+++ b/asia
@@ -2693,9 +2693,37 @@ Zone	Asia/Pyongyang	8:23:00 -	LMT	1908 Apr  1
 # Lebanon
 #
 # From Saadallah Itani (2023-03-23):
-# Lebanon too announced today delay of Spring forward from March 25 to April 20.
-# From Paul Eggert (2023-03-23):
+# Lebanon ... announced today delay of Spring forward from March 25 to April 20.
+#
+# From Paul Eggert (2023-03-27):
+# This announcement was by the Lebanese caretaker prime minister Najib Mikati.
 # https://www.mtv.com.lb/en/News/Local/1352516/lebanon-postpones-daylight-saving-time-adoption
+# A video was later leaked to the media of parliament speaker Nabih Berri
+# asking Mikati to postpone DST to aid observance of Ramadan, Mikati objecting
+# that this would cause problems such as scheduling airline flights, to which
+# Berri interjected, "What flights?"
+#
+# The change was controversial and led to a partly-sectarian divide.
+# Many Lebanese institutions, including the education ministry, the Maronite
+# church, and two news channels LCBI and MTV, ignored the announcement and
+# went ahead with the long-scheduled spring-forward on March 25/26, some
+# arguing that the prime minister had not followed the law because the change
+# had not been approved by the cabinet.  Google went with the announcement;
+# Apple ignored it.  At least one bank followed the announcement for its doors,
+# but ignored the announcement in internal computer systems.
+# Beirut international airport listed two times for each departure.
+# Dan Azzi wrote "My view is that this whole thing is a Dumb and Dumber movie."
+# Eventually the prime minister backed down, said the cabinet had decided to
+# stick with its 1998 decision, and that DST would begin midnight March 29/30.
+# https://www.nna-leb.gov.lb/en/miscellaneous/604093/lebanon-has-two-times-of-day-amid-daylight-savings
+# https://www.cnbc.com/2023/03/27/lebanon-in-two-different-time-zones-as-government-disagrees-on-daylight-savings.html
+#
+# Although we could model the chaos with two Zones, that would likely cause
+# more trouble than it would cure.  Since so many manual clocks and
+# computer-based timestamps ignored the announcement, stick with official
+# cabinet resolutions in the data while recording the prime minister's
+# announcement as a comment.  This is how we treated a similar situation in
+# Rio de Janeiro in spring 1993.
 #
 # Rule	NAME	FROM	TO	-	IN	ON	AT	SAVE	LETTER/S
 Rule	Lebanon	1920	only	-	Mar	28	0:00	1:00	S
@@ -2719,11 +2747,12 @@ Rule	Lebanon	1988	only	-	Jun	1	0:00	1:00	S
 Rule	Lebanon	1989	only	-	May	10	0:00	1:00	S
 Rule	Lebanon	1990	1992	-	May	1	0:00	1:00	S
 Rule	Lebanon	1992	only	-	Oct	4	0:00	0	-
-Rule	Lebanon	1993	2022	-	Mar	lastSun	0:00	1:00	S
+Rule	Lebanon	1993	max	-	Mar	lastSun	0:00	1:00	S
 Rule	Lebanon	1993	1998	-	Sep	lastSun	0:00	0	-
 Rule	Lebanon	1999	max	-	Oct	lastSun	0:00	0	-
-Rule	Lebanon	2023	only	-	Apr	21	0:00	1:00	S
-Rule	Lebanon	2024	max	-	Mar	lastSun	0:00	1:00	S
+# This one-time rule was announced by the prime minister but soon withdrawn.
+#Rule	Lebanon	2023	only	-	Apr	21	0:00	1:00	S
+
 # Zone	NAME		STDOFF	RULES	FORMAT	[UNTIL]
 Zone	Asia/Beirut	2:22:00 -	LMT	1880
 			2:00	Lebanon	EE%sT
diff --git a/tz-link.html b/tz-link.html
index 3b5fe770..43190dd7 100644
--- a/tz-link.html
+++ b/tz-link.html
@@ -282,7 +282,7 @@ community, and data distributors downstream.
 If your government plans to change its time zone boundaries or
 daylight saving rules, please send email to <a
 href="mailto:tz@iana.org"><code>tz@iana.org</code></a> well in advance,
-as this will coordinate updates to many cell phones,
+as this will lessen confusion and will coordinate updates to many cell phones,
 computers, and other devices around the world.
 In your email, please cite the legislation or regulation that specifies
 the change, so that it can be checked for details such as the exact times
@@ -311,8 +311,8 @@ which means they will continue to use out-of-date rules.</p>
 <p>
 For these reasons any rule change should be promulgated at least a
 year before it affects how clocks operate; otherwise, there is a good
-chance that many clocks will operate incorrectly after the change, due
-to delays in propagating updates to software and data.
+chance that many clocks will be wrong due to delays in propagating updates,
+and that residents will be confused or even actively resist the change.
 The shorter the notice, the more likely clock problems will arise; see "<a
 href="https://codeofmatt.com/2016/04/23/on-the-timing-of-time-zone-changes/">On
 the Timing of Time Zone Changes</a>" for examples.
-- 
2.37.2

